src/Build/BackEnd/Components/RequestBuilder/TaskBuilder.cs

```diff
@@ -420,6 +420,11 @@ private async Task<WorkUnitResult> ExecuteBucket(TaskHost taskHost, ItemBucket b
                     if (requirements != null)
                     {
                         TaskLoggingContext taskLoggingContext = _targetLoggingContext.LogTaskBatchStarted(_projectFullPath, _targetChildInstance);
+                        if (taskLoggingContext.TaskName == "MSBuild")
```

Comment:
It's possible for tasks other than `MSBuild` to cause other projects to build--any task that (like MSBuild itself) calls [`IBuildEngine.BuildProjectFile`](https://github.com/Microsoft/msbuild/blob/d3ae07ed0ca30ed03861215786135f9f9a699cd7/src/Framework/IBuildEngine.cs#L79-L101) or [`IBuildEngine3.BuildProjectFilesInParallel`](https://github.com/Microsoft/msbuild/blob/d3ae07ed0ca30ed03861215786135f9f9a699cd7/src/Framework/IBuildEngine3.cs#L15-L42) could be the reason here.

Let me think about the consequences of changing your `CurrentMSBuildTask` to something set more generically.

Reply:
Please, much appreciated.

Reply:
I'd argue that while this only covers the MSBuild task scenario and not the IBuildEngine APIs, there's so much value in this change that it's worth adding now, and not worrying about the remaining scenarios, which I think should be quite, if not extremely, rare.